src/Component/Ssh/Client.php

```diff
@@ -70,6 +71,8 @@ public function run(Host $host, string $command, array $config = [])
             $terminalOutput($type, $buffer);
         };
 
+        $command = $this->replacePlaceholders($command, $config['vars']);
```

Comment:
It will be cool to have tests here as well. Please git pull and see test/joy/OnceTest for example.

Reply:
I think the problem is that even with using `ApplicationTester`, the tests are still being executed using `\ProcessRunner` instead of the `Ssh\Client`.

To test it with unit tests I would either:
* have to rework the `Ssh\Client` class, replace the `Process::fromShellCommandline` with some kind of "injectable" factory - this would allow to mock the Process class and inspect what the Client is passing there.
* create a very small environment that the tests would run in (based on docker-compose for instance).

What do you think?

Reply:
Yes, you are right.

> onvert it to "end-to-end" test by creating a very small environment that this test would run in (based on docker-compose for instance).

Adding end-to-end will really awesome.

PS Take a look at f429c750bc95dd153228257a3967039661b0bb40